source/boundary_temperature/interface.cc

```diff
@@ -341,6 +353,16 @@ namespace aspect
                            "implemented in a plugin in the BoundaryTemperature "
                            "group, unless an existing implementation in this group "
                            "already provides what you want.");
+        prm.declare_entry ("Allow fixed temperature on outflow boundaries", "true",
+                           Patterns::Bool (),
+                           "When the temperature is fixed on a given boundary as determined "
+                           "by the list of 'Fixed temperature boundary indicators', there "
+                           "might be parts of the boundary where material flows out and "
+                           "one may want to prescribe the temperature only on the parts of "
+                           "the boundary where there is inflow. This parameter determines "
+                           "if temperatures are only prescribed at these inflow parts of the "
+                           "boundary (if false) or everywhere on a given boundary, independent "
+                           "of the flow direction (if true).");
```

Comment:
here add something similar, except that it is necessary to state that mathematically it is allowed to prescribe the temperature even on outflow boundaries *as long as the diffusion coefficient is nonzero*. In practice, however, this would only make physical sense if the diffusion coefficient is actually quite large to prevent the creation of a boundary layer.